### DISCUSSION TITLE
Question about normal OneDrive versus Sharepoint

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#1856] [Q&A] [answered]
Hi there! Love the client and have it working well, but one query I wondered someone could offer some comments on...

What's the actual difference between syncing a normal OneDrive folder and syncing/connecting to a Sharepoint?

I have one of each enabled/configured... but can't actually see any difference between behaviour on them? Is there some added functionality for Sharepoint?

Thanks!

**Comment by abraunegg** [comment, marked as answer]
@johnsidoli 
Thanks for your feedback - greatly appreciated.

In terms of differences between OneDrive (Business and Personal) and SharePoint - there are a number of differences because of how SharePoint was developed by Microsoft.

The most impactful issue is that the file that you upload, gets modified by SharePoint post upload. Additional metadata gets added and there is no way to turn that off on SharePoint. This means the file size + file hash is different from what you have locally, compared to what is now online, which is why the application will then 'true up' your local state with this updated file.

The client handles this SharePoint file modification issue so that you dont see any difference in behaviour. You can see the difference in behaviour if you increase your application logging verbosity.

Please read https://github.com/OneDrive/onedrive-api-docs/issues/935 and the Microsoft response https://github.com/OneDrive/onedrive-api-docs/issues/935#issuecomment-448450261 in regards to this Microsoft SharePoint ~bug~ feature.
